### PR TITLE
Serialize debug request into url hash after sending. This allows peop…

### PR DESCRIPTION
**Pull request by anuraaga**
…le to share urls that go straight to a pre-populated debug form and it's response.

Links are just /docs/#path?req=<url encoded json as is>

The links can get long due to lots of url encoding, but maybe it's ok since it's the fragment. I could also do a base64 encoding to shorten them if it seems useful.

### CODE REVIEW

File: src/main/resources/com/linecorp/armeria/server/docs/assets/js/armeria.js [hunk 1]
```diff
@@ -19,21 +21,30 @@ $(function () {
...
+      // Path hasn't changed so don't re-render (probably just updated link
+      // for sharing debug requests).
+      return;
+    }
+    previousFragmentPath = fragmentUri.pathname();
+
     resetClasses();
 
-    var hashSplit = hash.split('/');
+    var hashSplit = fragmentUri.pathname().split('/');
```
Comment (delegacy) [inline src/main/resources/com/linecorp/armeria/server/docs/assets/js/armeria.js]: `fragmentUri.pathname()` appears three times. Assign a local variable for this?

File: src/main/resources/com/linecorp/armeria/server/docs/assets/js/armeria.js [hunk 2]
```diff
@@ -106,15 +117,27 @@ $(function () {
...
+    };
+    functionContainer.find('.debug-submit').on('click', submitDebugRequest);
 
     functionContainer.removeClass('hidden');
+
+    var fragment = URI(window.location.href).fragment(true);
+    var fragmentParams = fragment.query(true);
+    if (fragmentParams.req) {
+      debugText.val(fragmentParams.req);
+      submitDebugRequest();
```
Comment (delegacy) [inline src/main/resources/com/linecorp/armeria/server/docs/assets/js/armeria.js]: Submit by default?

**Comment by delegacy** [comment]
Left some minor comments. :smile:

File: src/main/resources/com/linecorp/armeria/server/docs/assets/js/armeria.js (hunk 2 above)
Comment (anuraaga) [inline src/main/resources/com/linecorp/armeria/server/docs/assets/js/armeria.js]: Yeah - I feel the link is to the actual response and submitting by default is good behavior. Imagine a QA report with a debug link inside it - you click it to see what the response was to debug in pretty much all cases.

**Comment by anuraaga** [comment]
Thanks - addressed / replied

File: src/main/resources/com/linecorp/armeria/server/docs/assets/js/armeria.js (hunk 2 above)
Comment (delegacy) [inline src/main/resources/com/linecorp/armeria/server/docs/assets/js/armeria.js]: I see your point. :)

**Comment by delegacy** [comment]
:shipit:

**Comment by trustin** [comment]
:+1:

File: src/main/resources/com/linecorp/armeria/server/docs/assets/js/armeria.js (hunk 2 above)
Comment (trustin) [inline src/main/resources/com/linecorp/armeria/server/docs/assets/js/armeria.js]: But someone could send someone a malicious link?

**Pull request merged**